mm2src/mm2_event_stream/src/streamer.rs

```diff
@@ -0,0 +1,232 @@
+use std::any::{self, Any};
+
+use crate::{Event, StreamingManager};
+use common::executor::{abortable_queue::WeakSpawner, AbortSettings, SpawnAbortable};
+use common::log::{error, info};
+
+use async_trait::async_trait;
+use futures::channel::{mpsc, oneshot};
+use futures::{select, FutureExt, Stream, StreamExt};
+
+/// A marker to indicate that the event streamer doesn't take any input data.
+pub struct NoDataIn;
```

Comment:
`type NoDataIn = ()` should be preferred here.

Reply:
Having a specific marker is better here to distinguish between a simple notification message (with no data) and the case where the streamer actually accepts no notification from KDF runtime.

Reply:
> Having a specific marker is better here to distinguish between a simple notification message (with no data) and the case where the streamer actually accepts no notification from KDF runtime.

`()` means void, isn't that already distinguished enough as no message uses this?

Reply:
nope, some message(s) use it